CMakeLists.txt

```diff
@@ -29,20 +32,63 @@ endif()
 # Set a default build type if none was specified
 set(default_build_type "Release")
 
+#
+# Diagnostic options
+#
+option(AT_DIAG_ENABLE_TEXT "Enable textural diagnostics to stderr" ON)
+option(AT_DIAG_ENABLE_DEBUG "Enable debug level diagnostics (chatty)" OFF)
+option(AT_DIAG_ENABLE_ASSERT "Enable asserts" ON)
+option(AT_DIAG_ENABLE_TRACING "Enable tracing of the detector algorithm" ON)
+option(AT_DIAG_ENABLE_TIMESTAMPS "Enable profiling of the detector through timestamps" ON)
+option(AT_DIAG_MATH "Enable math diagnostics" OFF)
+
+#
+# Developer debug flags
+#
+option(AT_DEBUG_GAUSSIAN "Debug the gaussian blur kernel" OFF)
+
+#
+# Developer test flags (experimental features)
+#
+option(AT_TEST_THRESHOLD_DEGLITCH "Enable threshold deglitching for noisy images" OFF)
+
+option(AT_HOMOGRAPHY_METHOD_INVERSE "Use inverse method" ON)
+cmake_dependent_option(AT_HOMOGRAPHY_METHOD_CHO_OR_PLU "Use CHO matrix solver" OFF AT_HOMOGRAPHY_METHOD_INVERSE CHO)
```

Comment:
Can you move those up to the other build options? Let's keep all the options in one place, so one has not to search through all of the CMakeLists.txt.

Reply:
How do you feel if I group options together and then the triggering of those options (e.g. ASAN, if(ASAN)) separately? We can either have the option and its trigger together or all options then all triggers.

Reply:
I would prefer to have the options grouped together. So when someone opens the `CMakeLists.txt` to look for the options, they only have to scroll to one section.